source/dpp/translation/macro_.d

```diff
@@ -566,3 +550,80 @@ private auto fixCasts(R)(
 
     return chain(beginning, middle, tokens[lastIndex .. $]);
 }
+
+/// Returns: `token index` for the matching closing parentheses. If no
+/// matching closing parentheses is found, `-1` is returned.
+private ptrdiff_t findMatchingParentheses(
+    const(from!"clang".Token)[] tokens,
+    ptrdiff_t i
+    )
+    @safe
+in (tokens[i].isPunctuation("("))
+out (r; r == -1
+    || tokens[r].isPunctuation(")")) //" quote is just here to unconfuse my syntax highlighter
+{
+    int depth = 1;
+    while (depth != 0) {
+        ++i;
+        if (i >= tokens.length)
+            break;
+        if(tokens[i].isPunctuation("("))
+            ++depth;
+        if(tokens[i].isPunctuation(")"))
+            --depth;
+    }
+
+    if (i >= tokens.length)
+        return -1;
+    return i;
+}
+
+///
+@("findMatchingParentheses")
+@safe unittest
```

Comment:
Why are you testing private functions?

Reply:
unittests for the internal functions that are newly added with this PR. Also they serve as documentation for the developers inside this module. In case they get made public in the future, everything is there.